### PR TITLE
Reduce access log noise from frontend paths

### LINKED ISSUE #4095: Access log records all /console/ and /gate/ frontend traffic
### Description

The HTTP access log records a line for **every request served by the `/console/` and `/gate/` frontend handlers**, not just API/protocol traffic. 

A single Console or Gate page load fetches the HTML shell and several assets, and each one produces its own access log line, drowning out the API and protocol traffic.

Example noise from a single Console page load:

 ```
GET /console/ HTTP/1.1 200 4820 1 ebdaf92d-1f43-4231-bd12-1ade05d0f58d" trace_id=ebdaf92d-1f43-4231-bd12-1ade05d0f58d
```
```
GET /console/config.js HTTP/1.1 200 1481 0 c21fbb53-bc72-46c9-b5c4-c4ac193302cd" trace_id=c21fbb53-bc72-46c9-b5c4-c4ac193302cd
```
```
GET /console/assets/index-CDGxQ6RE.js HTTP/1.1 200 255949 11 791174bb-69ae-4a30-af42-764745a42f1a" trace_id=791174bb-69ae-4a30-af42-764745a42f1a
```
```
GET /console/assets/vendor-mui-D-p82nfq.js HTTP/1.1 200 411820 8 26daeb26-b7b3-4bea-9f79-3d1d1e04bbf5" trace_id=26daeb26-b7b3-4bea-9f79-3d1d1e04bbf5
```
```
GET /console/flows/signin/{{ meta(application.logoUrl) }} HTTP/1.1 200 4820 0 bab502b0-c22d-4274-ab6c-c0325ebd872f" trace_id=bab502b0-c22d-4274-ab6c-c0325ebd872f  
```
and the same for `/gate/`

### Steps to Reproduce

1. Build and run ThunderID.
2. Open the Console at `https://localhost:8090/console` and sign in (or just hit any `/console/` or `/gate/` route).
3. Observe the server access log.
4. Note the many INFO access log lines for `GET /console/`, `/console/config.js`, `/console/assets/*.js`, `/console/flows/...`, and the equivalent `/gate/...` lines.

### Version

0.48.0

### Environment Details (with versions)

- OS: macOS (arm64), distribution `thunderid-0.48.0-macos-arm64`
- Database: default SQLite

### PR DESCRIPTION
**Pull request by Osara-B**
### Purpose

The access log recorded a line for every HTTP request, including requests for the static frontend assets served under `/console/` and `/gate/`. Loading either app fetches many JS, CSS, and image files, so a single page load produced a burst of access log lines that were not useful for observing API traffic. This made the log noisy and harder to read.

This PR lets the access log skip configured path prefixes so that static frontend routes no longer emit access log lines, keeping the log focused on API traffic.
Fixes #4095

### Approach
- `AccessLogHandler` now takes a `skipPrefixes []string` argument. When a request path starts with any of the given prefixes, the request is served normally but no access log line is written. All other requests are logged exactly as before.
- The server wires this up in `createHTTPServer` with `/gate/` and `/console/` as the skipped prefixes.
- Prefix matching uses the trailing slash deliberately, so only assets under those apps are skipped. Bare paths such as `/console` (the redirect to `/console/`) and unrelated paths that merely share a leading substring (for example `/consolexyz` or `/gateway`) are still logged.
- Added unit tests covering the skip behaviour: skipped paths are still served but produce no log line, non-skipped API paths are logged as usual, and the edge cases above (bare prefix, shared-substring paths, regular API traffic) behave correctly. Extracted the repeated test remote address into a shared constant.

### Related Issues
- https://github.com/thunder-id/thunderid/issues/4095

### Checklist
- [x] Followed the contribution guidelines.
- [ ] Manual test round performed and verified.
- [ ] Documentation provided. (Add links if there are any)
    - [ ] Ran Vale and fixed all errors and warnings
- [x] Tests provided. (Add links if there are any)
    - [x] Unit Tests
    - [ ] Integration Tests
- [ ] Breaking changes. (Fill if applicable)
    - [ ] Breaking changes section filled.
    - [ ] `breaking change` label added.

### Security checks
- [x] Followed secure coding standards in [WSO2 Secure Coding Guidelines](https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/introduction/)
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.

## Summary by CodeRabbit

- **Bug Fixes**
  - Reduced unnecessary access-log entries by skipping access logging for `/gate/` and `/console/`, plus any additional prefixes configured via `log.access.exclude_paths`.
  - Skipped requests are still served normally.
  - Strengthened prefix matching to avoid incorrect skips on edge cases.

- **New Features**
  - Added configurable “Access Log” exclusion paths via `log.access.exclude_paths`.

- **Tests**
  - Added tests verifying skipped paths emit no access-log output and that non-skipped routes continue to log, including edge-case coverage.

- **Documentation**
  - Documented the new **Access Log** configuration with an example excluding `/health/`.

### CODE REVIEW

File: backend/cmd/server/main.go [hunk 1]
```diff
@@ -233,7 +233,8 @@ func createHTTPServer(ctx context.Context, logger *log.Logger, cfg *config.Confi
 	// Build the middleware chain with proper execution order.
 	// Request flow: CorrelationID (outermost) -> AccessLog -> Security -> Route Handler (innermost)
 	// Note: Middlewares are wrapped in reverse order - the last added will execute first.
-	handler := log.AccessLogHandler(logger, securityMiddleware)
+	// Skip access logging for static frontend routes to keep the log focused on API traffic.
+	handler := log.AccessLogHandler(logger, []string{"/gate/", "/console/"}, securityMiddleware)
```
[the remark this answers is not in the document]
Comment (Osara-B) [reply backend/cmd/server/main.go]: Documentation not required

Comment (senthalan) [inline backend/cmd/server/main.go]: We are currently skipping the /healthcheck endpoint too. can we align that also to this pattern

Comment (Osara-B) [reply backend/cmd/server/main.go]: Added in [Reduce access log noise from frontend paths](https://github.com/thunder-id/thunderid/pull/4097/commits/3ce2aa7befebe5ca8737e3c39a23b5a0bb7747a7)

Comment (senthalan) [reply backend/cmd/server/main.go]: Ah. I thought we are already skipping the /health endpoint access log. Seems like we are not. There might be a requirement to see whether the health check endpoint have reached the server. So shall we move these urls under a configuration?

Comment (Osara-B) [reply backend/cmd/server/main.go]: Added in [Make access-log path exclusions configurable](https://github.com/thunder-id/thunderid/pull/4097/commits/962c8523f6345c5ab390f2edd801d28acf8af777). 

A guard is added against "" and  "/" , since `exclude_paths` uses prefix matching, and it would match every request and disable all access logging.

**Review by senthalan — APPROVED** [review]
